src/python/pants/subsystem/subsystem.py

```diff
@@ -169,3 +174,33 @@ def get_options(self):
     :API: public
     """
     return self._scoped_options
+
+  @staticmethod
+  def get_streaming_workunit_callbacks(subsystem_names: List[str]) -> List[Callable]:
```

Comment:
The parameter could be `Iterable` or `Sequence`, although it's good that the return type is `List`. For parameter types, we generally want to be flexible when possible for nicer call sites. But for return types, we want to be as precise as possible to improve MyPy's type inference for the results of calling functions.